src/main/scala/com/fulcrumgenomics/vcf/filtration/EndRepairArtifactLikelihoodFilter.scala

```diff
@@ -0,0 +1,212 @@
+      }
+    }
+
+    // Apply a prior based on the MAF
+    val maf = (altGood + altBad) / (altGood + altBad + refGood + refBad).toDouble
+    val priorMutation = min(pow(maf * 2, 2), 1)
+    val priorArtifact = 1 - priorMutation
+    llMutation += LnProb.toLogProbability(priorMutation)
+    llArtifact += LnProb.toLogProbability(priorArtifact)
+
+    val total = LnProb.or(llMutation, llArtifact)
+    val pMutation = expProb(LnProb.normalizeByLogProbability(llMutation, total))
+    val pArtifact = expProb(LnProb.normalizeByLogProbability(llArtifact, total))
+
+    Map(InfoKey -> pMutation)
+  }
+
+  /** Calculates a pair of priors:
+    *   The first is the prior that the genotype is the result of a true somatic mutations
+    *   The second is the prior that the genotype is the result of the end-repair artifact process
+    *
+    * Default implementation gives a mutation prior of {{{(maf * 2)^2}}} capped at 0.9999 and an
+    * artifact prior of 1 - mutation prior.  This has the effect of creating priors that are heavily
+    * skewed towards the artifact at very low MAFs, and towards mutations as the MAF increases towards 0.5.
+    */
+  private[filtration] def priors(pileup: Pileup[PileupEntry], gt: Genotype, maf: Double) : (Double, Double) = {
+    require(maf > 0 && maf <= 1, "maf must be greater than 0 and no more than 1.")
+    val priorMutation = min(pow(maf * 2, 2), 0.9999)
+    val priorArtifact = 1 - priorMutation
+    (priorMutation, priorArtifact)
+  }
+
+  /** Exponentiates a log probability that should be in the range 0-1 and caps it there to account for numerical imprecision. */
+  private def expProb(lnP: LnProb): Double = {
+    val x = exp(lnP)
+    if (x > 1) 1 else if (x < 0) 0 else x
+  }
+
+  /**
+    * Returns true if a base in the pileup (either ref or alt) is congruent with the artifact.
+    * I.e. the base is within the first or last few bases of the template _and_ the base as read
+    * makes sense given the position.
+    */
+  private[filtration] def isArtifactCongruent(refAllele: Byte, altAllele: Byte, entry : BaseEntry, pileupPosition: Int): Boolean = {
+    require(entry.base == refAllele || entry.base == altAllele, "Pileup base is neither ref or alt.")
+    val isRef                = entry.base == refAllele
+    val positionInRead       = entry.positionInReadInReadOrder
+    val positionFromOtherEnd = positionFromOtherEndOfTemplate(entry.rec, pileupPosition)
+
+    // Pick the position that's closest to and end, and define the "bad" base by that end
+    val (pos, badAlt) = {
+      if (positionFromOtherEnd.exists(_ < positionInRead)) (positionFromOtherEnd.get, 'A')
+      else                                                 (positionInRead, 'T')
+    }
+    val badRef  = if (badAlt == altAllele) refAllele else SequenceUtil.complement(refAllele)
+    val badBase = if (isRef) entry.baseInReadOrientation == badRef else entry.baseInReadOrientation == badAlt
```

Comment:
I also think the complementing makes it very confusing.  Can this all be done in read order from the start?  I think then it would be much clearer along with an explanation for how the artifact works.